pkg/kube/krt/helpers.go

```diff
@@ -57,13 +59,34 @@ func GetKey[O any](a O) string {
 	if ok {
 		return arn.ResourceName()
 	}
+	auid, ok := any(a).(uidable)
+	if ok {
+		return strconv.FormatUint(uint64(auid.uid()), 10)
+	}
+
+	akclient, ok := any(a).(kube.Client)
+	if ok {
+		return string(akclient.ClusterID())
+	}
+
 	ack := GetApplyConfigKey(a)
 	if ack != nil {
 		return *ack
 	}
 	panic(fmt.Sprintf("Cannot get Key, got %T", a))
 }
 
+// GetKey returns the key for the provided object.
+// If there is none, this will panic.
+func GetKey[O any](a O) string {
+	acluster, ok := any(a).(config.ObjectWithCluster[O])
```

Comment:
why not just implement ResourceName on ObjectWithCluster?

Reply:
That would require a type constraint on ObjectWithCluster's T; not everything is necessarily going to be a resource namer, though I think everything we're practically using today will be once I remove some cruft